executor-scripts/linux/vrrp

```diff
@@ -0,0 +1,358 @@
+			echo "Please configure main interface ${IFACE} with VRRP ID ${VRRP_ID} before adding IP addresses to VRRP interface."
+			continue
+		    fi
+		    if [ -z "${MOCK}" -a ! -d /sys/class/net/${IFACE_VRRP_6} ]; then
+			echo "Please configure main interface ${IFACE} with VRRP ID ${VRRP_ID} before adding IP addresses to VRRP interface."
+			continue
+		    fi
+		fi
+	    else
+		# This should by an IPv4 or IPv6 address
+		# Check if we have VRRP_ID in a previously step. Maybe somehow an invalid VRRP ID escape from checks
+		if ! is_vrid $VRRP_ID; then
+		    echo "Invalid VRRP ID: $VRRP_ID"
+		    VRRP_ID=0
+		    continue
+		fi
+
+		VRRP_IP_ADDR=${VRRP_IP%*/*}
+		VRRP_PREFIX=${VRRP_IP#*/*}
+
+		# Check if IP address has an netmask. If not, considering /24 or /64
+		if [ -z "$VRRP_PREFIX" ]; then
+		    if [ addr_family ${VRRP_IP_ADDR} -eq 6 ]; then
+			VRRP_NETMASK=64
+		    else
+			VRRP_NETMASK=24
+		    fi
+		fi
+
+		# Check if NETMASK if is a valid one: IPv4 between 1 to 32 and IPv6 1 to 128.
+		if [ $(addr_family ${VRRP_IP_ADDR}) -eq 6 ]; then
+		    if [ ${VRRP_PREFIX} -lt 1 -o ${VRRP_PREFIX} -gt 128 ]; then
+			echo "Invalid IPv6 prefix ($VRRP_PREFIX). Valid prefixes are between 1 and 128."
+			continue
+		    else
+			# Add IP address to the VRRP interface.
+			${MOCK} ip -6 address add ${VRRP_IP_ADDR}/${VRRP_PREFIX} dev ${IFACE_VRRP_6}
+		    fi
+		else
+		    if [ ${VRRP_PREFIX} -lt 1 -o ${VRRP_PREFIX} -gt 32 ]; then
+			echo "Invalid IPv4 prefix ($VRRP_PREFIX). Valid prefixes are between 1 and 32."
+			continue
+		    else
+			# Add IP address to the VRRP interface.
+			${MOCK} ip address add ${VRRP_IP_ADDR}/${VRRP_PREFIX} dev ${IFACE_VRRP_4}
+		    fi
+		fi
+		
+		#VRRP_ID=$(echo $VRRP_IP | cut -f1 -d",")
+		#VRRP_ADDRESSES=$(echo $VRRP_IP | cut -f2- -d"," | sed "s/,/ /g")
+
+		#VRRP_ID_HEX=$(printf "%02x" $VRRP_ID)
+		#IFACE_VRRP="v${IFACE_CRC32}-${VRRP_ID_HEX}-"
+
+		#for VRRP_ADDR in ${VRRP_ADDRESSES}; do
+		#    if [ "${VRRP_ADDR##*/*}" ] ;then
+		#	if [ addr_family $VRRP_ADDR -eq 6 ]; then
+		#	    VRRP_MASK=64
+		#	else
+		#	    VRRP_MASK=24
+		#	fi
+		#    fi
+		#done
+	    fi
+	done
+	;;
+
+post-up)
+	;;
+pre-down)
+	init_vrrp
+
+	IFACE_CRC32=$(compute_crc32_hex ${IFACE})
+
+	if [ -z "${MOCK}" -a ! -d "/sys/class/net/${IFACE}/" ]; then
+	    exit 0
+	fi
+	if [ -z "${MOCK}" ]; then
+	    VRRP_IFACES=`find /sys/class/net/${IFACE}/ -name "upper_v${IFACE_CRC32}-*" -printf "%f\n" | sed "s/upper_//g"`
+	else
+	    # Let's generate list of the interfaces for MOCK. In test mode, these interfaces doesn't exist in system.
+	    VRRP_IFACES=""
+	    for VRRP_ID in ${IF_VRRP_ID}; do
+		if ! is_vrid $VRRP_ID; then
+		    continue
+		fi
+	    # Compute VRRP interfaces name
+	    vrrp_gen_name
+	    VRRP_IFACES="$VRRP_IFACES "${IFACE_VRRP_4}" "${IFACE_VRRP_6}
+	    done
+	fi
+
+	# VRRP interfaces should be destroied in post-down, becase if you have a bond, bridge or teaming interfaces, these are destroied in "destroy" stage.
```

Comment:
```suggestion
	# VRRP interfaces should be destroyed in post-down, because if you have a bond, bridge or teaming interfaces, these are destroyed in "destroy" stage.
```